lib/src/values/logic_value.dart

```diff
@@ -679,6 +679,43 @@ abstract class LogicValue {
   // ignore: avoid_dynamic_calls
   LogicValue operator %(dynamic other) => _doMath(other, (a, b) => a % b);
 
+  /// Unary clog2 operation.
+  ///
+  /// Returns ceil of log base 2 of valid input.
+  /// Returns `x` if any bit of input is invalid.
+  LogicValue clog2() => _doUnaryMath(_clog2);
+
+  /// Executes mathematical operations on single [LogicValue]
+  ///
+  /// Handles invalid input and do proper type conversion to required types
+  LogicValue _doUnaryMath(dynamic Function(dynamic num) op) {
+    if (!isValid) {
+      return LogicValue.filled(width, LogicValue.x);
+    }
+    if (this is BigInt ||
+        this is _BigLogicValue ||
+        (this is _FilledLogicValue && width > _INT_BITS)) {
+      final num = toBigInt();
+      return LogicValue.ofBigInt(op(num) as BigInt, width);
+    } else {
+      final num = toInt();
+      return LogicValue.ofInt(op(num) as int, width);
+    }
+  }
+
+  /// Ceil of log base 2 Operation
+  ///
+  /// Input [num] will be either of type [int] or [BigInt].
+  /// Returns ceil of log base 2 of [num]  having same type as of input.
+  dynamic _clog2(dynamic num) {
+    if (num is int) {
```

Comment:
Does this perform correctly for `int`s that can be interpreted as negative numbers?  Perhaps we need to cast to BigInt all the way up to `_INT_BITS-1` width?  Or else perform some other special consideration?

Reply:
Yes, I did test a few values and cross checked online with $clog2 (with 64 bits) which is working correctly as per clog2 implemented. Actual $clog2 always converts the negative input to an equivalent positive number using the standard way (2^bits + (input negative number)). Hence input is always a positive equivalent.

The implemented clog2 does the same.

E.g., `LogicValue.ofInt(-128, 8).clog2()` outputs 7
         `LogicValue.ofInt(-127, 8).clog2()` outputs 8

Works similarly for BigInt.

I will add negative cases also if we need to proceed with PR.

Reply:
The case I was thinking of was negative numbers which are at the int-width boundary, since that's the point where they get interpreted as negative by the int libraries that use `bitLength`.  For example, it seem this test is failing:
```dart
expect(LogicValue.ofInt(-1, 64).clog2().toInt(), 64);
```